src/operator/every.ts

```diff
@@ -8,38 +8,41 @@ import {Subscriber} from '../Subscriber';
 import {tryCatch} from '../util/tryCatch';
 import {errorObject} from '../util/errorObject';
 
+export type Predicate<T> = (value: T, index: number, source: Observable<T>) => boolean;
+
 /**
  * Returns an Observable that emits whether or not every item of the source satisfies the condition specified.
  * @param {function} predicate a function for determining if an item meets a specified condition.
  * @param {any} [thisArg] optional object to use for `this` in the callback
  * @returns {Observable} an Observable of booleans that determines if all items of the source Observable meet the condition specified.
  */
-export function every<T>(predicate: (value: T, index: number, source: Observable<T>) => boolean,
-                         thisArg?: any): Observable<boolean> {
-  const source = this;
-  if (source._isScalar) {
-    const result: boolean = tryCatch(predicate).call(thisArg || this, source.value, 0, source);
+export function every<T>(predicate: Predicate<T>, thisArg?: any): Observable<boolean> {
+    
+  let _this: Observable<T> = this;
```

Comment:
I hadn't discovered `source` until I got to this operator (and I'm not sure there's more than two others where it is used), so I didn't realize this was the idiom for this codebase. I have no strong opinions on the name used however, and `source` is actually much nicer.